src/Client.php

```diff
@@ -30,6 +31,17 @@ public function __construct($uri, $options, $driverOptions)
     }
 
     /**
+     * Drop a database.
+     *
+     * @param string $databaseName
+     * @return Result
+     */
+    public function dropDatabase($databaseName)
```

Comment:
The proto should definetly not be string|Database, it should be string - but if it does (string) casting inside the method it implicitly allows the Database object to work seamlessly.

as for dropCollection() returning fully qualified namespace...... mmmmh.....